src/engine/currencyEngine.js

```diff
@@ -586,9 +596,13 @@ export class CurrencyEngine {
         address => scriptHashes[address]
       )
 
+      const randomArray = this.io.random(32)
+      const b = Buffer.from(randomArray)
```

Comment:
It would be a lot simpler to just use an incrementing counter. Each id would then be unique without needing randomness.

Reply:
Good point. I'll update the PR